cmd/upgrade/app/v1alpha2/0.9.0-to-1.0.0/jiva/jiva_upgrade.go

```diff
@@ -0,0 +1,387 @@
+	pvClient      = pv.NewKubeClient(pv.WithKubeConfigPath(kubeConfigPath))
+)
+
+func getDeploymentDetails(labels, namespace string) (
+	deployName,
+	containerName,
+	version string,
+	err error,
+) {
+	deployList, err := deployClient.WithNamespace(namespace).List(
+		&metav1.ListOptions{
+			LabelSelector: labels,
+		})
+	if err != nil {
+		return "", "", "", err
+	}
+	if len(deployList.Items) == 1 {
+		deployName = deployList.Items[0].Name
+		version = deployList.Items[0].Labels["openebs.io/version"]
+		if deployName == "" {
+			return "", "", "", errors.New("empty deployment name")
+		}
+		if len(deployList.Items[0].Spec.Template.Spec.Containers) == 0 {
+			return "", "", "", errors.New("empty container list")
+		}
+		containerName = deployList.Items[0].Spec.Template.Spec.Containers[0].Name
+		if containerName == "" {
+			return "", "", "", errors.New("empty container name")
+		}
+	} else {
+		return "", "", "", errors.New("deployment missing")
+	}
+	return deployName, containerName, version, nil
+}
+
+func patchDelpoyment(
+	deployName,
+	namespace string,
+	pt types.PatchType,
+	data []byte,
+) error {
+	var (
+		retries = 60
+	)
+	_, err := deployClient.WithNamespace(namespace).Patch(
+		deployName,
+		pt,
+		data,
+	)
+	if err != nil {
+		return err
+	}
+	for {
+		retries = retries - 1
+		rolloutStatus, err := deployClient.WithNamespace(namespace).
+			RolloutStatus(deployName)
+		if err != nil {
+			return err
+		}
+		time.Sleep(5 * time.Second)
+		if retries == 0 || rolloutStatus.IsRolledout {
+			if !rolloutStatus.IsRolledout {
+				return errors.Errorf("failed to rollout %s", rolloutStatus.Message)
+			}
+			break
+		}
+	}
+	return nil
+}
+
+func getBaseImage(deployObj *appsv1.Deployment, name string) (string, error) {
+	for _, con := range deployObj.Spec.Template.Spec.Containers {
+		if con.Name == name {
+			return strings.Split(con.Image, ":")[0], nil
+		}
+	}
+	return "", errors.Errorf("image not found for %s", name)
+}
+
+func main() {
+	// inputs required for the upgrade
+	upgradeVersion := "1.0.0"
+	currentVersion := "0.9.0"
+	pvName := "pvc-e9c1b919-aa19-11e9-bea9-54e1ad5e8320"
+	openebsNamespace := "openebs"
+
+	var (
+		pvLabel         = "openebs.io/persistent-volume=" + pvName
+		replicaLabel    = "openebs.io/replica=jiva-replica," + pvLabel
+		controllerLabel = "openebs.io/controller=jiva-controller," + pvLabel
+		ns              string
+		err             error
+	)
+
+	// verifying whether the pvc is deployed with DeployInOpenebsNamespace cas config
+	deployInOpenebs, err := deployClient.WithNamespace(openebsNamespace).List(
+		&metav1.ListOptions{
+			LabelSelector: pvLabel,
+		})
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+	if len(deployInOpenebs.Items) > 0 {
+		ns = openebsNamespace
+	} else {
+		pvObj, err := pvClient.Get(pvName, metav1.GetOptions{})
+		if err != nil {
+			fmt.Println("namespace missing")
+			os.Exit(1)
+		}
+		ns = pvObj.Spec.ClaimRef.Namespace
+	}
```

Comment:
I am not able to understand above logic of namespace extraction. In other words, can we write this in a better way?